packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -81,6 +83,14 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 					continue;
 				}
 
+				const context = {};
+
+				if ( source.usesContext?.length ) {
+					for ( const key of source.usesContext ) {
+						context[ key ] = blockContext[ key ];
+					}
+				}
```

Comment:
Worth noting that we do the exact same for blocks basically: 

https://github.com/WordPress/gutenberg/blob/f5ee54fa7a48ceb3b03140fdd23097a636431275/packages/block-editor/src/components/block-edit/edit.js#L56-L64